### PR TITLE
Move to pyproject.toml

### PR DESCRIPTION
**Pull request by vfreex**
Also needs to add `build` PyPI package to the CI image: https://github.com/openshift/ocp-build-data-validator/pull/88

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    "tenacity",
+    "jira >= 3.4.1",  # https://github.com/pycontribs/jira/issues/1486
+]
+
+[project.scripts]
+elliott = "elliottlib.cli.__main__:main"
+
+[project.urls]
+homepage = "https://github.com/openshift/elliott"
+repository = "https://github.com/openshift/elliott.git"
```
Comment (thegreyd) [inline pyproject.toml:69]:
```suggestion
homepage = "https://github.com/openshift-eng/elliott"
repository = "https://github.com/openshift-eng/elliott.git"
```

File: requirements.txt
Comment (thegreyd) [inline requirements.txt:1]: Can you update the Makefile as well?

Comment (vfreex) [reply requirements.txt:1]: done

**Pull request merged**